front/src/applications/operationalStudies/components/SimulationResults/SpeedSpaceChart/d3Helpers.js

```diff
@@ -208,7 +209,106 @@ function drawTrain(
         rotate
       );
     }
+    if (dataSimulation.modesAndProfiles && speedSpaceSettings.electricalProfiles) {
+      dataSimulation.modesAndProfiles.forEach((source, index) => {
+        // prepare object to work with
+        const segment = {};
 
+        segment.position_start = source.start;
+        segment.position_end = source.stop;
+        segment.position_middle = (source.start + source.stop) / 2;
+        segment.lastPosition = dataSimulation.modesAndProfiles.slice(-1)[0].stop;
+        segment.height_start = 4;
+        segment.height_end = 24;
+        segment.height_middle = (segment.height_start + segment.height_end) / 2;
+        segment.usedMode = source.used_mode;
+        segment.usedProfile = source.used_profile;
+
+        // prepare colors
```

Comment:
Please set these consts elsewhere outside the context

Reply:
done